.travis.yml

```diff
@@ -3,18 +3,33 @@ matrix:
     - os: windows
       language: sh
       python: 3.6
-      name: Build on Python 3.6.8 in Windows
-      env: PYVER="3.6.8" CAIRO_VERSION=1.17.2
+      name: Build on x64 Python 3.6.8 in Windows
+      env: ARCH=x64 CAIRO_VERSION=1.17.2 PYTHON_PACKAGE=python PYTHON_VERSION="3.6.8"
     - os: windows
       language: sh
       python: 3.7
-      name: Build on Python 3.7.7 in Windows
-      env: PYVER="3.7.7" CAIRO_VERSION=1.17.2 
+      name: Build on x64 Python 3.7.7 in Windows
+      env: ARCH=x64 CAIRO_VERSION=1.17.2 PYTHON_PACKAGE=python PYTHON_VERSION="3.7.7" 
     - os: windows
       language: sh
       python: 3.8
-      name: Build on Python 3.8.3 in Windows
-      env: PYVER="3.8.3" CAIRO_VERSION=1.17.2
+      name: Build on x64 Python 3.8.3 in Windows
+      env: ARCH=x64 CAIRO_VERSION=1.17.2 PYTHON_PACKAGE=python PYTHON_VERSION="3.8.3"
```

Comment:
This can be updated to `3.8.5` as it is released now.
```suggestion
      env: ARCH=x64 CAIRO_VERSION=1.17.2 PYTHON_PACKAGE=python PYTHON_VERSION="3.8.5"
```

Reply:
Good spot...

We can automate this, I spoke to dependencies.io who gave me an account for open source stuff + that can generate diffs and PRs for this sort of thing. 

It's annoying nuget doesn't accept a query like 3.7.x and give the highest match, I wonder if chocolatey can do it?

Reply:
You can use a query like `3.7` I think.

Reply:
Also I just found that choco and nuget are one and the same thing in structure. Choco uses nugets structure.

Reply:
Chocolatey has some extensions, I'm inclined to switch back to it, as it has some advantages.
Native support in Travis: one less thing to install, and Travis' cache support seems to include it. 

Semver: not supported now, but they have a ticket.
https://github.com/chocolatey/choco/issues/1610
 
In the meantime I think we might be able to do semver support with a combination of powershell and the output of `choco search` or hit their API directly with `cURL`. 

This would be good as we could specify python version with 2 digit versions, not 3, so one less thing to support and patch.

Reply:
> You can use a query like `3.7` I think.
This doesn't work directly right now, so you get `3.7.0` :( but yeah, we can defintely use a query like that... I'm having a play with the choco commandline to see if I can persaude it to list all versions, not much luck yet - but you can see on the site they exist, so there will be a way of doing it somehow.

Reply:
So.. .discovered some things.   There is [a bug](https://github.com/chocolatey/choco/issues/1843) in current chocolatey (fixed in next version) that prevents this kind of search.

But in the meantime it's possible to downgrade chocolatey:

`choco install chocolatey -y --version 0.10.13 --allow-downgrade`

Then we can search all versions:

```shell
$ choco search -e python3 -a
Chocolatey v0.10.13
python3 3.9.0-rc1 [Approved]
python3 3.9.0-b5 [Approved]
python3 3.9.0-b4 [Approved]
[snipped a lot of versions]
```

Use grep to search for python 3.7 and remove the [Approved] column:
```shell
$ choco search -e python3 -a | grep 'python3 3.7' | head | cut -d' ' -f1,2
python3 3.7.9
python3 3.7.8
python3 3.7.7
python3 3.7.6.20200110
python3 3.7.6
python3 3.7.5
python3 3.7.4
python3 3.7.3
python3 3.7.2
python3 3.7.2-rc1
```

^ It looks like they come back in the order they were added, so which seems like what we want.

Get exact python version given "3.7":
```
choco search -e python3 -a | grep 'python3 3.7' | head | cut -d' ' -f2 | head -1
3.7.9
```

Reply:
I really don't know what to tell on this. But yeah I prefer using choco but does it has x86 python? I could find one so I use nuget for it and was also in [python docs](https://docs.python.org/3/using/windows.html#windows-nuget). Think about it before using. Also, we can use nuget feed in choco by

```powershell
choco install python -source="https://www.nuget.org/api/v2"
```
This will `nuget`  feed instead of chocolatey's.

Reply:
We can pass  - -x86 to install 32 bit versions, I haven't tried it yet.

Reply:
That doesn't help much. Nuget python is build for cli ,but choco is using default py installer which may be troublesome. So better to use nuget for x86.

Reply:
This can be easily fixed if we were to use GitHub Actions.